test/e2e/podtimingcontroller/podtimingcontroller.go

```diff
@@ -61,30 +61,43 @@ func (r *podTimingReconciler) reconcile(ctx context.Context, req ctrl.Request) e
 		return err
 	}
 
-	phase := pod.Status.Phase
-	podPhaseIdentifier := string(phase) + pod.Namespace + pod.Name + string(pod.UID)
+	if !isPodReady(&pod) {
+		return nil
```

Comment:
so if the pod never gets ready we don't record anything, is that ok?

Reply:
I think so, the purpose of this controller is really just to track how long it takes until all the controlplane components come up. It is also possible for a pod to get ready and then not ready again, which is not going to show up here at all.

I was hoping that we verify elsewhere that all controlplane components get ready eventually, do we?

Reply:
we kinda do through our semantic success criteria which is covey in the hcp.status, e.g version, release, ControlPlaneEndpoint, kubeconfig, conditions...